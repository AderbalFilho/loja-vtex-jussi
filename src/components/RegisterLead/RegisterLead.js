import React from 'react';

import { JButton, JInput, JTitle, JSection } from '@ui';

import './RegisterLead.scss';

function RegisterLead() {
  function submit(e) {
    e.preventDefault();
    // Register lead
  }

  return (
    <JSection color="black" classes="register-lead">
      <div className="register-lead__container">
        <div className="register-lead__titles">
          <JTitle tag="h2" color="green">
            &#47;&#47;
          </JTitle>
          <JTitle tag="h2" color="white">
            receba novidades da nossa Ã¡rea de produtos digitais.
          </JTitle>
        </div>
        <form onSubmit={submit} className="register-lead__register">
          <JInput
            type="text"
            placeholder="Digite seu e-mail"
            classes="register-lead__input"
            classesContainer="register-lead__input-container"
          />
          <JButton type="submit" color="green" classes="register-lead__button">
            Cadastrar
          </JButton>
        </form>
      </div>
    </JSection>
  );
}

export default RegisterLead;
