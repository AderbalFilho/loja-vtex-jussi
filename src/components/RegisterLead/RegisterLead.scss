.register-lead {
  padding: 136px 0;
}

.register-lead__container {
  margin: 0 auto;
  max-width: 535px;
  width: 100%;
}

.register-lead__titles {
  position: relative;
  text-align: center;

  h2 {
    margin: 0 0 24px 0;

    &:first-child {
      margin: 0 0 0 -48px;
      position: absolute;
    }
  }
}

.register-lead__register {
  border-bottom: 1px solid var(--green-jussi);
  width: 502px;
}

.register-lead__input {
  background-color: var(--black);
  border: none;
  color: var(--white);

  &::placeholder {
    color: var(--white);
  }
}

.register-lead__input-container {
  display: inline-block;
  margin-left: -16px;
  width: 404px;
}

.register-lead__button {
  display: inline-block;
  padding-right: 0;
  text-align: right;
  text-transform: uppercase;
}
