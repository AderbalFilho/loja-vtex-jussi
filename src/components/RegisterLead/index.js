import RegisterLead from './RegisterLead';

export default RegisterLead;
