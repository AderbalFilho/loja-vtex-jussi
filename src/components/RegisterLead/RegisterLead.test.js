import React from 'react';
import TestRenderer from 'react-test-renderer';

import RegisterLead from './RegisterLead';

describe('RegisterLead', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<RegisterLead />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
