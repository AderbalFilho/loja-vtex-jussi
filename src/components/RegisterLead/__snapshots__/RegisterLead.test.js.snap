// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterLead matches snapshot 1`] = `
<section
  className="j-section j-section--black register-lead"
>
  <div
    className="register-lead__container"
  >
    <div
      className="register-lead__titles"
    >
      <h2
        className="j-title j-title--green j-title--large"
      >
        //
      </h2>
      <h2
        className="j-title j-title--white j-title--large"
      >
        receba novidades da nossa Ã¡rea de produtos digitais.
      </h2>
    </div>
    <form
      className="register-lead__register"
      onSubmit={[Function]}
    >
      <div
        className="j-input register-lead__input-container"
      >
        <input
          className="j-input__input register-lead__input"
          onChange={[Function]}
          placeholder="Digite seu e-mail"
          type="text"
          value=""
        />
        
      </div>
      <button
        className="j-button j-button--back-transparent j-button--border-black j-button--green j-button--rad-normal register-lead__button"
        type="submit"
      >
        Cadastrar
      </button>
    </form>
  </div>
</section>
`;
